Tests/DatadogTests/Datadog/URLSessionAutoInstrumentation/Interception/URLSessionInterceptorTests.swift

```diff
@@ -29,6 +29,8 @@ class URLSessionInterceptorTests: XCTestCase {
     private let handler = URLSessionInterceptionHandlerMock()
     /// Mock request made to a first party URL.
     private let firstPartyRequest = URLRequest(url: URL(string: "https://api.first-party.com/v1/endpoint")!)
+    /// Alternative mock request made to a first party URL.
+    private let alternativeFirstPartyRequest = URLRequest(url: URL(string: "https://api.another-first-party.com/v1/endpoint")!)
```

Comment:
We don't cover this behaviour in tests:
* _if one set of hosts is defined in global config, but another set of hosts is specified for `DDURLSessionDelegate`, the global set is not considered first party._

I think it deserves separate test function.

Reply:
we don't cover many other things too, in theory we need to duplicate every test case we currently have with this new configuration but that would be too much.
as an attempt to keep a balance, i tested new config in very basic cases only; shall we change that?